api/tests.py

```diff
@@ -41,7 +41,7 @@ def test_access_phone_list_unauthenticated(self, mock):
         # check that we have a drf-tracking log entry
         self.assertEqual(APIRequestLog.objects.count(), 1)
         self.assertIn(self.ok_user.phone, APIRequestLog.objects.first().data)
-        self.assertEqual(APIRequestLog.objects.first().response, "")
+        self.assertNotEqual(APIRequestLog.objects.first().response, "")
```

Comment:
```suggestion
        self.assertContains(APIRequestLog.objects.first().response, self.ok_user.email)
```

checks that the response data was logged. 


Would also be good to check that the response contained the data also. So on line 40 add something like

```
self.assertContains(response, self.ok_user.email)
```

Reply:
I was about to write too that this test anyways is not really the actual 481 test, that what should one really do in this part? Actual test is on line 199 or so (test_access_phone_notok)

Reply:
Also, neither suggestion work in that location anyways...

`self.assertContains(response, self.ok_user.email)`

```
FAIL: test_access_phone_list_unauthenticated (api.tests.TestLogging)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/mock.py", line 1204, in patched
    return func(*args, **keywargs)
  File "/home/olmari/sorsat/mulysa/api/tests.py", line 40, in test_access_phone_list_unauthenticated
    self.assertContains(response, self.ok_user.email)
  File "/home/olmari/.local/share/virtualenvs/mulysa-HIB61Lec/lib/python3.7/site-packages/django/test/testcases.py", line 462, in assertContains
    response, text, status_code, msg_prefix, html)
  File "/home/olmari/.local/share/virtualenvs/mulysa-HIB61Lec/lib/python3.7/site-packages/django/test/testcases.py", line 434, in _assert_contains
    " (expected %d)" % (response.status_code, status_code)
AssertionError: 481 != 200 : Couldn't retrieve content: Response code was 481 (expected 200)
```

`self.assertContains(APIRequestLog.objects.first().response, self.ok_user.email)`

```
ERROR: test_access_phone_list_unauthenticated (api.tests.TestLogging)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/mock.py", line 1204, in patched
    return func(*args, **keywargs)
  File "/home/olmari/sorsat/mulysa/api/tests.py", line 44, in test_access_phone_list_unauthenticated
    self.assertContains(APIRequestLog.objects.first().response, self.ok_user.email)
  File "/home/olmari/.local/share/virtualenvs/mulysa-HIB61Lec/lib/python3.7/site-packages/django/test/testcases.py", line 462, in assertContains
    response, text, status_code, msg_prefix, html)
  File "/home/olmari/.local/share/virtualenvs/mulysa-HIB61Lec/lib/python3.7/site-packages/django/test/testcases.py", line 432, in _assert_contains
    response.status_code, status_code,
AttributeError: 'str' object has no attribute 'status_code'
```

Reply:
Ah yeah assertContains checks the response code also. Use assertIn (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn)

Reply:
..not like it didn't have an example right there at the same place for data part. new eyes, new code =)